src/main/java/gtPlusPlus/xmod/gregtech/common/covers/CoverOverflowValve.java

```diff
@@ -0,0 +1,373 @@
+package gtPlusPlus.xmod.gregtech.common.covers;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.ForgeDirection;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.google.common.io.ByteArrayDataInput;
+import com.gtnewhorizons.modularui.api.screen.ModularWindow;
+import com.gtnewhorizons.modularui.common.widget.TextWidget;
+
+import gregtech.api.gui.modularui.CoverUIBuildContext;
+import gregtech.api.gui.modularui.GTUITextures;
+import gregtech.api.interfaces.metatileentity.IMetaTileEntity;
+import gregtech.api.interfaces.tileentity.ICoverable;
+import gregtech.api.metatileentity.CommonMetaTileEntity;
+import gregtech.api.metatileentity.implementations.MTEBasicTank;
+import gregtech.api.metatileentity.implementations.MTEFluid;
+import gregtech.api.util.CoverBehaviorBase;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.ISerializableObject;
+import gregtech.common.gui.modularui.widget.CoverDataControllerWidget;
+import gregtech.common.gui.modularui.widget.CoverDataFollowerNumericWidget;
+import gregtech.common.gui.modularui.widget.CoverDataFollowerToggleButtonWidget;
+import io.netty.buffer.ByteBuf;
+
+public class CoverOverflowValve extends CoverBehaviorBase<CoverOverflowValve.OverflowValveData> {
+
+    private final int minOverflowPoint = 0;
+    private final int maxOverflowPoint;
+
+    public CoverOverflowValve(int maxOverflowPoint) {
+        super(OverflowValveData.class);
+        this.maxOverflowPoint = maxOverflowPoint;
+    }
+
+    @Override
+    public OverflowValveData createDataObject() {
+        return new OverflowValveData(0, 0, false, false);
+    }
+
+    @Override
+    public OverflowValveData createDataObject(int aLegacyData) {
+        return new OverflowValveData(0, 0, false, false);
+    }
+
+    private FluidStack doOverflowThing(FluidStack fluid, OverflowValveData data) {
+        if (fluid != null && fluid.amount > data.overflowPoint)
+            fluid.amount -= Math.min(fluid.amount - data.voidingRate, data.voidingRate);
+        return fluid;
+    }
+
+    private void doOverflowThings(FluidStack[] fluids, OverflowValveData data) {
+        for (FluidStack fluid : fluids) doOverflowThing(fluid, data);
+    }
+
+    @Override
+    protected OverflowValveData doCoverThingsImpl(ForgeDirection side, byte aInputRedstone, int aCoverID,
+        OverflowValveData data, ICoverable aTileEntity, long aTimer) {
+        if (data == null) return new OverflowValveData(0, 0, false, false);
+
+        if (data.overflowPoint == 0) return data;
```

Comment:
You shouldn't prevent this cover from performing action when overflow point is set to zero. Its totally valid case - to void tank completely. Instead you should prevent any actions if voiding rate is 0. 
```suggestion
        if (data.voidingRate == 0) return data;
```

Reply:
I didn't think about that, thanks for the advice, I'll think through the logic better in the future if I change/add anything else

Reply:
> You shouldn't prevent this cover from performing action when overflow point is set to zero. Its totally valid case - to void tank completely. Instead you should prevent any actions if voiding rate is 0.

I didn't think about that, thanks for the advice, I'll think through the logic better in the future if I change/add anything else

Reply:
Done in commit https://github.com/GTNewHorizons/GT5-Unofficial/pull/3040/commits/3c1274c905452cea324eb5e411f1e99ead798933